src/guides/v2.3/test/static/static_test_execution.md

```diff
@@ -0,0 +1,55 @@
+---
+group: testing
+title: Running Static Tests
+functional_areas:
+  - Testing
+  - test
+---
+
+Executing Magento 2 static tests is straightforward. They can be executed in several ways.
+
+### Running static tests on all files
+
+To run static tests on all files, navigate to the Magento base directory and execute the following command:
+
+```bash
+bin/magento dev:test:run static
```

Comment:
@gabrieldagama: out of curiosity, are these the exact same tests that are being ran when you submit a PR on github?
Unless something has changed recently that's not the case. It seems like there are a bunch of extra static tests which are being executed on github.
[I'm looking for easy to comprehend information](https://github.com/magento/devdocs/issues/4495) about how to execute all these static tests on your local machine, it will save so much waiting time when working on a PR if we have this information.

Reply:
Hi @hostep, this is a valid point, thanks for bringing it up! Yes, they are the same tests, but they run a little bit differently, on our automated tests, the Static Tests are split in their separate test suite respectively (Less, HTML, PHP, Integrity...) and when you run locally, it runs all the suites at once. We should get the same results locally.

But there are somethings that I've noticed:
- Some of the tests are run based on changed files (code style, for example), so if you commit your changes before running the tests you may not get the errors.
- We do have some differences in EE and B2B versions, so if you don't have those installed you may get different results.

Maybe we can add this information to the docs as well, what do you think?

Hope this clarifies a little bit your question. If you have some PR that we can look at to see the differences it would be great!

Reply:
Ok thanks for the info!

Ah yes, I seem to remember that it was almost always some extra tests in B2B that failed. Unfortunately we are only using Open Source so I have no access to those extra tests.

What is the reason for the static tests to be different on EE & B2B projects btw? These are static tests, in my opinion they should be 100% the same between the different type of Magento projects. I understand there is a difference for integration & functional tests, but I don't understand why there is a difference for static tests?
Would be great if the differences in static tests could get aligned so they are the same over all types of Magento projects, no?

Reply:
I agree with you, from what I can see the PHP check are exactly the same, it just changes some files that are ignored or not. But, there are some differences in other static tests, like Integrity and Legacy tests. I'm fully aware of exactly the reasons and what are the differences, but I may be able to check internally.

But yeah, from my point of view, ideally we would have a defined test strategy that is publicly available and applied to all versions.

Reply:
I also seem to remember vaguely that php mess detector static test ran on B2B and not on OS, but I might be mistaken